web/src/routes/EventsPage.tsx

```diff
@@ -81,10 +82,19 @@ const EventsPage = ({ city, pathname, languageCode, cityCode }: CityRouteProps):
     Toolbar: <CityContentToolbar slug={event?.slug} />,
   }
 
-  if (!events && loading) {
+  if (loading) {
     return (
       <CityContentLayout isLoading {...locationLayoutParams}>
-        <LoadingSpinner />
+        {!event && !eventId ? (
```

Comment:
🤔 Shouldn't the check for `eventId` be enough? Also, please don't use negations.